torchrl/envs/transforms/transforms.py

```diff
@@ -3607,3 +3607,56 @@ def transform_input_spec(self, input_spec: CompositeSpec) -> CompositeSpec:
             if not self.create_copy:
                 del input_spec[in_key]
         return input_spec
+
+
+class Reward2GoTransform(Transform):
+    """Calculates the reward to go based on the episode reward and a discount factor.
+
+    As the Reward2GoTransform is only an inverse transform the in_keys will be directly used for the in_keys_inv.
+    The reward-to-go can be only calculated once the episode is finished. Therefore, the transform should be applied to the replay buffer
+    and not to the collector.
+
+    Args:
+        in_keys (list of str/tuples of str): the entries to rename
+        gamma (float or torch.Tensor): the discount factor
+
+    """
+
+    def __init__(
+        self,
+        gamma: Optional[Union[float, torch.Tensor]] = 1.0,
+        in_keys: Optional[Sequence[str]] = None,
+    ):
+        if in_keys is None:
+            in_keys = ["reward"]
+        out_keys = ["reward_to_go"]
+        super().__init__(
+            in_keys=in_keys,
+            in_keys_inv=in_keys,
+            out_keys_inv=out_keys,
+        )
+
+        if not isinstance(gamma, torch.Tensor):
+            gamma = torch.tensor(gamma)
+
+        self.register_buffer("gamma", gamma)
+
+    def _inv_call(self, tensordict: TensorDictBase) -> TensorDictBase:
+        episode_ends = torch.where(tensordict.get(("next", "done")))[0]
```

Comment:
we can also have `("next", "truncated")`.

```suggestion
        done = tensordict.get(("next", "done"))
        truncated = tensordict.get(("next", "truncated"), None)
        if truncated is not None:
            done_or_truncated = done | truncated
        else:
            done_or_truncated = done
        if not done_or_truncated.any(-2).all():
            raise RuntimeError("No episode ends found")
        episode_ends = torch.where(done)[0]
```